theano/gpuarray/dnn.py

```diff
@@ -2732,6 +2732,189 @@ def local_abstractconv3d_cudnn_graph(op, context_name, inputs, outputs):
+
+    __props__ = ()
+    _cop_num_inputs = 3
+    _cop_num_outputs = 1
+
+    def __init__(self):
+        DnnBase.__init__(self, ["c_code/spatialtf_grid.c"], "spatialtf_grid")
+
+    def dnn_context(self, node):
+        return node.outputs[0].type.context_name
+
+    def make_node(self, grid_dimensions, theta, desc, precision, cx=None):
+        if cx is None:
+            context_name = infer_context_name(desc, theta)
+        else:
+            context_name = infer_context_name(desc, theta, cx)
+
+        grid_dimensions = as_tensor_variable(grid_dimensions)
+        theta = as_gpuarray_variable(theta, context_name)
+
+        # Allocate GPU memory for grid of coordinates
+        grid = GpuArrayType(dtype=precision,
+                            broadcastable=(False, False, False, False,),
+                            context_name=context_name)()
+
+        return Apply(self, [theta, grid_dimensions, desc], [grid])
+
+    def L_op(self, inputs, outputs, output_grads):
+        pass
+
+
+class GpuDnnGridSampler(DnnBase):
+
+    """
+    This Op builds a spatial transformer grid sampler for use in spatial transformer network
+    operations.
+    """
+
+    __props__ = ()
+    _cop_num_inputs = 6
+    _cop_num_outputs = 1
+
+    def __init__(self):
+        DnnBase.__init__(self, ["c_code/spatialtf_sampler.c"], "spatialtf_sampler")
+
+    def dnn_context(self, node):
+        return node.outputs[0].type.context_name
+
+    def make_node(self, img, output, grid, grid_dimensions, desc,
+                  alpha=None, beta=None, cx=None):
+        if cx is None:
+            context_name = infer_context_name(img, grid)
+        else:
+            context_name = infer_context_name(img, grid, cx)
+
+        img = as_gpuarray_variable(img, context_name)
+        output = as_gpuarray_variable(output, context_name)
+        grid = as_gpuarray_variable(grid, context_name)
+        grid_dimensions = as_tensor_variable(grid_dimensions)
+
+        if img.type.ndim != 4:
+            raise TypeError('img must be a 4D tensor')
+        if output.type.ndim != 4:
+            raise TypeError('output must be a 4D tensor')
+
+        if img.type.ndim != output.type.ndim:
+            raise TypeError('The number of dimensions of img and output must match')
+
+        if (not isinstance(desc.type, CDataType) or
+                desc.type.ctype != 'cudnnSpatialTransformerDescriptor_t'):
+            raise ValueError('desc must be cudnnSpatialTransformerDescriptor_t')
+
+        alpha = ensure_dt(alpha, _one, 'alpha', img.dtype)
+        beta = ensure_dt(beta, _zero, 'beta', img.dtype)
+
+        return Apply(self, [img, grid, grid_dimensions, desc, alpha, beta],
+                     [output.type()])
+
+    def L_op(self, inputs, outputs, output_grads):
+        pass
+
+
+def dnn_spatialtf(img, theta, grid_dims, alpha=None, beta=None, precision=None):
+    """
+        GPU spatial transformer using cuDNN from NVIDIA.
+    """
+    precision = theano.config.floatX
```

Comment:
I think this line should not be here, as precision should be given as function arg.